thriftpy/parser/parser.py

```diff
@@ -790,20 +211,22 @@ def _make_enum(name, kvs):
             val = -1
         for item in kvs:
             if item[1] is None:
-                item[1] = val + 1
-            val = item[1]
-        for key, val in kvs:
+                val = val + 1
+            else:
+                val = item[1]
+            key = item[0]
             setattr(cls, key, val)
             _values_to_names[val] = key
             _names_to_values[key] = val
-    setattr(cls, '_VALUES_TO_NAMES', _values_to_names)
-    setattr(cls, '_NAMES_TO_VALUES', _names_to_values)
+    cls.'_VALUES_TO_NAMES' = _values_to_names
+    cls.'_NAMES_TO_VALUES' = _names_to_values
     return cls
 
 
-def _make_empty_struct(name, ttype=TType.STRUCT, base_cls=TPayload):
-    attrs = {'__module__': thrift_stack[-1].__name__, '_ttype': ttype}
-    return type(name, (base_cls, ), attrs)
+def _make_empty_struct(name, module, ttype=TType.STRUCT, base_cls=TPayload):
+    attrs = {'__module__': module.__name__, '_ttype': ttype}
+    module.__dict__[name] = type(name, (base_cls, ), attrs)
+    return module.__dict__[name]
```

Comment:
i think `setattr`/`getattr` might be a bit more standard here, if i'm not missing anything